### PR TITLE
refactor: Switches from mocha to ava for test execution

### PR DESCRIPTION
**Pull request by KnisterPeter**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +38,7 @@
   },
   "homepage": "https://github.com/knisterpeter/paeckchen#readme",
   "devDependencies": {
+    "ava": "0.15.2",
```
Comment (marionebl) [inline package.json:41]: Do no forget to remove `mocha` on line 47

File: package.json [hunk 2]
```diff
@@ -9,18 +9,20 @@
     "src"
   ],
   "scripts": {
+    "#postinstall": "postinstall hook is required to 'fix' ava d.ts file",
+    "postinstall": "rimraf node_modules/ava/index.d.ts",
```
Comment (marionebl) [inline package.json]: What's wrong withe the ava typings?

Comment (KnisterPeter) [inline package.json:13]: The ava typings contain references to the promise object which is only available in typescript with compile to es2015.
paeckchen currently compiles to es5. Therefore the typings need to be patched.

File: package.json [hunk 3]
```diff
@@ -9,18 +9,20 @@
...
-    "test": "mocha --require source-map-support/register --recursive dist/test",
+    "test": "ava",
     "changelog": "conventional-changelog -p angular -i CHANGELOG.md -w",
     "preversion": "npm test",
     "version": "npm run build && conventional-changelog -p angular -v -r 0 > CHANGELOG.md && git add -A .",
     "postversion": "git push && git push --tags",
     "prepublish": "not-in-install && npm run build || echo 'In install cycle, skipping prepublish'",
     "precoverage": "npm run clean && tsc --inlineSourceMap",
-    "coverage": "nyc --reporter lcov ./node_modules/.bin/mocha --recursive dist/test --all"
+    "coverage": "nyc --reporter lcov ./node_modules/.bin/ava"
```
Comment (marionebl) [inline package.json]: Should work without `./node_modules/.bin/`

File: package.json (hunk 2 above)
Comment (marionebl) [inline package.json:13]: Neat idea with the comment for postinstall. You could rephrase to:
"Remove bundled typings to use patched version with Promise support"

Comment (marionebl) [inline package.json:13]: Fair enough. Couldn't we do this?
https://www.typescriptlang.org/docs/release-notes/typescript-1.8.html#augmenting-globalmodule-scope-from-modules

**Comment by marionebl** [comment]
LGTM. The noise resulting from typings removal/adding is nasty though.

**Comment by KnisterPeter** [comment]
Sure it is, but it will cool down when most dependencies are set ;)

**Comment by KnisterPeter** [comment]
And btw: Typings-Augmentation could only add elements to typings. In case of ava there are already references to Promises which we could not use (and must not define) in our setup.
Doing so will result in a compiler error.

**Pull request merged**